### PR TITLE
Make headers bold

### LINKED ISSUE #1228: Typography needs to be overhauled
Currently it's hard to find the right typo for your specific case. Some options aren't even there and some others one needs to find with trial and error, since this stuff is far from semantic.

I'm talking about this file: `20-typography-and-text.scss`

### LINKED ISSUE #1739: Heading: align breakpoints with AXA UI Kit
Heading has three breakpoints for font sizes, see:

```css
h1.a-heading {
  @include typo-title(large-1);

  @include breakpoint($mediaquery-md) {
    @include typo-title(huge-1);
  }

  @include breakpoint($mediaquery-lg-up) {
    @include typo-title(huge-3);
  }
```

But AXA UI Kit has only 2 breakpoints:

https://www.figma.com/file/6zurYk3bJpzUg0H2THSxGF/AXA-UI-Kit?node-id=0%3A58

image

This should be aligned: two breakpoints in UI Kit meas Heading should have two breakpoints also.
(as discussed with Jonas on 21.4.2020).

### LINKED ISSUE #1750: [Child of #1755] Typography: Heading and Text component
Feedback from @MKaHead.

After we merged the new mixins for teh typography, update those components accordingly and make sure to test it well.

### LINKED ISSUE #1796: Wrong font-weight on heading component
The font-weight for Source Sans Pro on the heading component is wrong. It should be bold instead of regular.

https://patterns.axa.ch/?path=/story/components-atoms-heading--heading

### PR DESCRIPTION
**Pull request by 01231**
Fixes #1796 fixes #1750 and fixes #1228 and fixes #1739

Done is when (DoD):
- [x] Modifications within components are reflected by tests.
- [x] A self-review of the code has been performed.
- [x] The modified component properties have been added to the typescript typings.
- [x] Potential design changes have been reviewed by a designer.
- [x] The modifications are well documented (README.md, etc.) and showcased in the stories.
- [x] The modifications are documented in the code, particularly in hard-to-understand areas (focus on **WHY**).
- [x] The modifications are shortly added to CHANGELOG.md with the issue number.
- [x] The modifications still work in IE.
- [x] The modifications are according to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)

### CODE REVIEW

File: src/components/10-atoms/heading/index.scss [hunk 1]
```diff
@@ -13,145 +13,49 @@
...
   &.a-heading--secondary {
-    @include typo-title(medium, secondary);
-
-    @include breakpoint($mediaquery-md) {
-      @include typo-title(medium-1, secondary);
-    }
-
-    @include breakpoint($mediaquery-lg-up) {
-      @include typo-title(large, secondary);
-    }
```
Comment (danyball) [inline src/components/10-atoms/heading/index.scss:155]: What happened to the breakpoints in this file? Ah... `text()` is doing that now?

File: src/components/10-atoms/button-link/index.scss [hunk 2]
```diff
@@ -21,7 +21,7 @@
   border-bottom-color: $color-deep-sapphire;
 
   @include breakpoint($mediaquery-sm-up) {
-    @include _text(default);
+    min-height: 40px;
```
Comment (danyball) [inline src/components/10-atoms/button-link/index.scss]: Why did you add min-height?

**Review by danyball — COMMENTED** [review]
Wow huge PR... good work- so much cleaner code now! I clicked around at the deployed branch... LGTM.

File: src/components/10-atoms/heading/index.scss (hunk 1 above)
Comment (01231) [reply src/components/10-atoms/heading/index.scss:155]: `headline()` has breakpoints for mobile and desktop

Comment (01231) [reply src/components/10-atoms/heading/index.scss:155]: `text()` has no breakpoints

File: src/components/10-atoms/button-link/index.scss (hunk 2 above)
Comment (NoahLiechti) [reply src/components/10-atoms/button-link/index.scss]: cant answer you...since the merge conflicts everything changed :(

Comment (01231) [reply src/components/10-atoms/button-link/index.scss]: There has nothing changed as you can see in the diffing  `src/components/10-atoms/button-link/index.scss`

File: src/components/10-atoms/button/index.scss [hunk 3]
```diff
@@ -19,8 +19,7 @@
   border: solid 2px $color-axa-blue;
   border-bottom-color: $color-deep-sapphire;
 
-  @include breakpoint($mediaquery-xs-up) {
-    @include _text(default);
+  @include breakpoint($mediaquery-sm-up) {
```
Comment (danyball) [inline src/components/10-atoms/button/index.scss]: You changed it to `sm-up`?

File: src/components/10-atoms/button/index.scss [hunk 4]
```diff
@@ -60,17 +59,17 @@
...
     min-height: 50px;
-    @include _text(medium);
+    @include text(t4);
   }
 }
 
 .a-button--small {
   min-height: 30px;
-  @include _text(small);
+  @include text(t4);
```
Comment (danyball) [inline src/components/10-atoms/button/index.scss:71]: text-small is `text(t4)` and text-medium is also t4 ?

File: src/components/10-atoms/input-file/index.scss [hunk 5]
```diff
@@ -57,11 +53,11 @@
 
 .a-input-file--large {
   min-height: 40px;
-  @include _text(default);
+  @include text(t4);
 
   @include breakpoint($mediaquery-xs-up) {
     min-height: 50px;
-    @include _text(medium)
+    @include text(t3);
```
Comment (danyball) [inline src/components/10-atoms/input-file/index.scss]: At this component is default -> t4 and medium -> t3. Thats different to button.

File: src/components/20-molecules/policy-features/policy-features-item/index.scss [hunk 6]
```diff
@@ -9,6 +9,7 @@
 
   @include breakpoint($mediaquery-sm-up) {
     width: 210px;
+
```
Comment (danyball) [inline src/components/20-molecules/policy-features/policy-features-item/index.scss]: remove empty line

File: src/components/30-organisms/commercial-hero-banner/child.scss [hunk 7]
```diff
@@ -17,56 +17,25 @@ axa-commercial-hero-banner {
     margin-right: 0;
   }
 
-  h2 {
```
Comment (danyball) [inline src/components/30-organisms/commercial-hero-banner/child.scss:20]: What happend to h2? Not needed?

**Review by danyball — CHANGES_REQUESTED** [review]

File: src/components/30-organisms/commercial-hero-banner/child.scss (hunk 7 above)
Comment (01231) [reply src/components/30-organisms/commercial-hero-banner/child.scss:20]: It's not a `<h2>` anymore. It's a `<p>` inside a `<header>`

File: src/components/10-atoms/button/index.scss (hunk 4 above)
Comment (01231) [reply src/components/10-atoms/button/index.scss:71]: button small was 13px in the past. We don't have that anymore. So I have to use the smallest available. This is `t4`

File: src/components/10-atoms/input-file/index.scss (hunk 5 above)
Comment (01231) [reply src/components/10-atoms/input-file/index.scss]: Input file `large` is broken #1838. I copied the text sizes from button now.

File: src/components/10-atoms/button/index.scss (hunk 3 above)
Comment (01231) [reply src/components/10-atoms/button/index.scss]: done

File: src/components/20-molecules/policy-features/policy-features-item/index.scss (hunk 6 above)
Comment (01231) [reply src/components/20-molecules/policy-features/policy-features-item/index.scss]: done

File: src/components/10-atoms/button/index.scss (hunk 4 above)
Comment (danyball) [reply src/components/10-atoms/button/index.scss:71]: And the size of t4 is the same as figma says for "button small"?

File: src/components/10-atoms/carousel/child.scss [hunk 8]
```diff
@@ -1,16 +1,16 @@
...
-  @include typo(primary);
+  @include text(t4);
   display: flex;
   flex-direction: column;
   text-align: center;
   justify-content: center; // align the dom element itself (f.e. small spans)
 
-  @include breakpoint($mediaquery-xs-up) {
-    @include typo(primary, large);
+  @include breakpoint($mediaquery-sm-up) {
```
Comment (danyball) [inline src/components/10-atoms/carousel/child.scss]: You changed the behaviour from xs to sm. Because the breakpoint also changed or a mistake?

File: src/components/10-atoms/checkbox/index.scss [hunk 9]
```diff
@@ -172,10 +172,10 @@ axa-checkbox {
   .a-checkbox__children-inline,
   .a-checkbox__children-inline * {
       display: inline;
-      @include typo(primary, small-1);
+      @include text(t4);
 
-      @include breakpoint($mediaquery-sm-up) {
-         @include typo(primary, medium);
+      @include breakpoint($mediaquery-md-up) {
```
Comment (danyball) [inline src/components/10-atoms/checkbox/index.scss]: You changed the behaviour from xs to sm. Because the breakpoint also changed or a mistake?

File: src/components/30-organisms/footer/child.scss [hunk 10]
```diff
@@ -20,8 +20,8 @@ axa-footer {
     text-transform: uppercase;
     line-height: 1.5;
 
-    @include breakpoint($mediaquery-xs-up) {
-      @include typo(primary, medium, bold);
+    @include breakpoint($mediaquery-sm-up) {
```
Comment (danyball) [inline src/components/30-organisms/footer/child.scss]: You changed the behaviour from xs to sm. Because the breakpoint also changed or a mistake?

File: src/components/30-organisms/footer/index.scss [hunk 11]
```diff
@@ -216,8 +216,8 @@
           border-bottom: none;
         }
 
-        @include breakpoint($mediaquery-xs-up) {
-          @include typo(primary, default, default);
+        @include breakpoint($mediaquery-sm-up) {
```
Comment (danyball) [inline src/components/30-organisms/footer/index.scss]: You changed the behaviour from xs to sm. Because the breakpoint also changed or a mistake?

File: src/components/30-organisms/footer/index.scss [hunk 12]
```diff
@@ -205,7 +205,7 @@
       background-color: $color-azure;
 
       &-list-item {
-        @include typo(primary, xx-small, default);
+        @include text(t4);
```
Comment (danyball) [inline src/components/30-organisms/footer/index.scss:208]: "xx-small" is different to t4 size. So... do it look good even with t4? Did you talk to a designer because of this design change?

File: src/components/30-organisms/testimonials/index.scss [hunk 13]
```diff
@@ -5,28 +5,18 @@
...
 
   &__subtitle {
-    @include typo();
+    @include text(t4);
     margin: 20px 30px 0 30px;
     text-align: center;
 
-    @include breakpoint($mediaquery-xs-up) {
-      @include typo(primary, medium);
+    @include breakpoint($mediaquery-sm-up) {
```
Comment (danyball) [inline src/components/30-organisms/testimonials/index.scss]: You changed the behaviour from xs to sm. Because the breakpoint also changed or a mistake?

**Review by danyball — COMMENTED** [review]
Maybe a designer should check some components of this PR?

File: src/components/10-atoms/button/index.scss (hunk 4 above)
Comment (01231) [reply src/components/10-atoms/button/index.scss:71]: t4 is 14px and figma says 14px for button small👍🏼

File: src/components/30-organisms/footer/index.scss (hunk 12 above)
Comment (01231) [reply src/components/30-organisms/footer/index.scss:208]: We have talked a lot about this. t4 is the smallest available so there is no other possibility. I have talked to a designer yesterday.

File: src/components/10-atoms/carousel/child.scss (hunk 8 above)
Comment (01231) [reply src/components/10-atoms/carousel/child.scss]: done

File: src/components/10-atoms/checkbox/index.scss (hunk 9 above)
Comment (01231) [reply src/components/10-atoms/checkbox/index.scss]: done

File: src/components/30-organisms/footer/child.scss (hunk 10 above)
Comment (01231) [reply src/components/30-organisms/footer/child.scss]: done

File: src/components/30-organisms/footer/index.scss (hunk 11 above)
Comment (01231) [reply src/components/30-organisms/footer/index.scss]: done

File: src/components/30-organisms/footer/index.scss (hunk 12 above)
Comment (01231) [reply src/components/30-organisms/footer/index.scss:208]: I don't know what xx-small was, but there is no visible change

File: src/components/30-organisms/testimonials/index.scss (hunk 13 above)
Comment (01231) [reply src/components/30-organisms/testimonials/index.scss]: done

**Review by danyball — APPROVED** [review]
